src/v/kafka/server/tests/usage_manager_test.cc

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/server/usage_aggregator.h"
+#include "storage/tests/kvstore_fixture.h"
+#include "vlog.h"
+
+#include <seastar/testing/thread_test_case.hh>
+#include <seastar/util/log.hh>
+
+static ss::logger af_logger{"test-accounting-fiber"};
+
+class test_accounting_fiber final
+  : public kafka::usage_aggregator<ss::manual_clock> {
```

Comment:
Very nice! It would be great to add some randomised tests here. You could randomly advance the timer, add data, restart the fibre and periodically check that the data is consistent with the expectations.

Reply:
There is one issue with this, can't really query the system clock , I can only infer what should occur between window close calls. I'm relying more on ducktape for the tests that deal with ensuring timestamps are correct and all